feature-libs/quote/components/testing/common-quote-test-utils.service.ts

```diff
@@ -0,0 +1,195 @@
+      0,
+      `expected element identified by selector '${querySelector}' to be present, but it is NOT! innerHtml: ${htmlElement.innerHTML}`
+    );
+  }
+
+  /**
+   * Helper function for verifying whether the expected number of elements is present in the HTML tree.
+   *
+   * @param {any} expect - Expectation for a spec.
+   * @param {Element} htmlElement - HTML element.
+   * @param {string} querySelector - Query selector.
+   * @param {number} numberOfElements - Number of elements.
+   */
+  static expectNumberOfElementsPresent(
+    expect: any,
+    htmlElement: Element,
+    querySelector: string,
+    numberOfElements: number
+  ) {
+    expect(htmlElement.querySelectorAll(querySelector).length).toBe(
+      numberOfElements,
+      `expected elements identified by selector '${querySelector}' to be present, but it is NOT! innerHtml: ${htmlElement.innerHTML}`
+    );
+  }
+
+  /**
+   * Helper function for verifying whether the element contains text.
+   *
+   * @param {any} expect - Expectation for a spec.
+   * @param {Element} htmlElement - HTML element.
+   * @param {string} querySelector - Query selector.
+   * @param {any} expectedText - Expected text.
+   * @param {number} index - Optional index of the element.
+   */
+  static expectElementToContainText(
+    expect: any,
+    htmlElement: Element,
+    querySelector: string,
+    expectedText: string,
+    index?: number
+  ) {
+    let text;
+    if (index) {
+      text = htmlElement.querySelectorAll(querySelector)[index]?.textContent;
+    } else {
+      text = htmlElement.querySelector(querySelector)?.textContent;
+    }
+    expect(text).toContain(expectedText);
+  }
+
+  /**
+   * Helper function for verifying whether the element is not present in the HTML tree.
+   *
+   * @param {any} expect - Expectation for a spec.
+   * @param {Element} htmlElement - HTML element.
+   * @param {string} querySelector - Query selector.
+   */
+  static expectElementNotPresent(
+    expect: any,
+    htmlElement: Element,
+    querySelector: string
+  ) {
+    expect(htmlElement.querySelectorAll(querySelector).length).toBe(
+      0,
+      `expected element identified by selector '${querySelector}' to be NOT present, but it is! innerHtml: ${htmlElement.innerHTML}`
+    );
+  }
+
+  /**
+   * Helper function for verifying whether the element contains attribute.
+   *
+   * @param {any} expect - Expectation for a spec.
+   * @param {Element} htmlElement - HTML element.
+   * @param {string} querySelector - Query selector.
+   * @param {string} attributeName - Name of the attribute.
+   * @param {number} index - Optional index of the element.
+   */
+  static expectElementToContainAttribute(
+    expect: any,
+    htmlElement: Element,
+    querySelector: string,
+    attributeName: string,
+    index?: number
+  ) {
+    const element = CommonQuoteTestUtilsService.getHTMLElement(
+      htmlElement,
+      querySelector,
+      index
+    );
```

Comment:
@Larisa-Staroverova: In order to reduce the number of method parameters we could simply provide the target element instead of the (root) htmlElement, query selector and index.
so that the caller first calls the getHTMLElement method and provides the result as parameter to this method.
```suggestion
  static expectElementToContainAttribute(
    expect: any,
    element: Element,
    attributeName: string,
  ) {
```
same holds true for all the other methods calling internally getHTMLElement

Reply:
the calling code would then look look like this
```
      const button = CommonQuoteTestUtilsService.getHTMLElement(htmlElem, '.btn:first-child');
      CommonQuoteTestUtilsService.expectElementToContainAttribute(expect, button, 'disabled');
```

Reply:
@Uli-Tiger, I don't agree, maybe the number of parameters will be reduced but the layout of the method will be changed. As you could see in this helper class all methods are written in the same way:
> expect: any,
> htmlElement: Element,
> querySelector: string
> ...

One could ask him/herself why  these methods (`expectElementNotToContainAttribute` and others) are implemented differently. Furthermore, one has to know / guess before calling this `expectElementNotToContainAttribute` method , `getHTMLElement` method should be called. 
The helper class was written in the first line to simplify the code and its readability, and in the second line to avoid code duplication but if we follow your suggestion then both will be missed.

Reply:
> One could ask him/herself why these methods (`expectElementNotToContainAttribute` and others) are implemented differently. Furthermore, one has to know / guess before calling, for instance this `expectElementNotToContainAttribute` method , `getHTMLElement` method should be called.

let me quote myself
>same holds true for all the other methods calling internally getHTMLElement

Reply:
@Larisa-Staroverova: I still find it confusing that we hand in parameter called `element` to a method named `expectElementToContainAttribute` but than search the attribute in a child element. 
**At least please name the parameter `rootElement` and the method `expectChildElementToContainAttribute`. The querySelector could be named `childElementQuerySelector` then.**

Reply:
Done!